add BFloat16 support on CPU for cumsum, cumprod and logcumsumexp and adjust grain size

mingfeima opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #76279
* __->__ #74899
* #74898
* #74897

Differential Revision: [D37441441](https://our.internmc.facebook.com/intern/diff/D37441441)

cc @jgong5 @XiaobingSuper @sanchitintel @ashokei @jingxu10

mingfeima committed:
add BFloat16 support on CPU for cumsum, cumprod and logcumsumexp and adjust grain size

[ghstack-poisoned]

mingfeima committed:
Update on "add BFloat16 support on CPU for cumsum, cumprod and logcumsumexp and adjust grain size"




[ghstack-poisoned]

mingfeima committed:
Update on "add BFloat16 support on CPU for cumsum, cumprod and logcumsumexp and adjust grain size"




[ghstack-poisoned]

frank-wei commented:
@frank-wei has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D37441441).

frank-wei commented:
Pls check the failing test cases

mingfeima committed:
Update on "add BFloat16 support on CPU for cumsum, cumprod and logcumsumexp and adjust grain size"


Differential Revision: [D37441441](https://our.internmc.facebook.com/intern/diff/D37441441)

[ghstack-poisoned]
